### PR TITLE
Add direct arg to login flow

### PR DESCRIPTION
**Pull request by PVince81**
Signed-off-by: Vincent Petry <vincent@nextcloud.com>

### CODE REVIEW

**Comment by PVince81** [comment]
- [ ] TODO: make it so that the "direct" argument does not get added if the value was not set at all

File: core/Controller/LoginController.php [hunk 1]
```diff
@@ -267,7 +267,12 @@ private function canResetPassword(?string $passwordLink, ?IUser $user): bool {
 
 	private function generateRedirect(?string $redirectUrl): RedirectResponse {
 		if ($redirectUrl !== null && $this->userSession->isLoggedIn()) {
-			$location = $this->urlGenerator->getAbsoluteURL($redirectUrl);
+			$location = null;
+			if (str_starts_with($redirectUrl, 'http')) {
```
[the remark this answers is not in the document]
Comment (CarlSchwan) [reply core/Controller/LoginController.php:271]: actually this code seems to be a bad idea security wise :( redirecting to a possible external URL is bad!

[the remark this answers is not in the document]
Comment (CarlSchwan) [reply core/Controller/LoginController.php:271]: I tested locally without this modification and it still works

**Comment by CarlSchwan** [comment]
needs to be moved master

**Comment by PVince81** [comment]
obsoleted by https://github.com/nextcloud/server/pull/31592

**Pull request closed without merging**